materialize-sql/validate.go

```diff
@@ -130,11 +130,8 @@ func checkTypeError(field string, existing *pf.CollectionSpec, proposed *pf.Coll
 		return "The proposed materialization is missing the projection, which is required because it's included in the existing materialization"
 	}
 
-	// Ensure that the possible types of the proposed are a subset of the existing possible types.
-	// The new projection is allowed to contain fewer types than the original, though, since that
-	// will always work with the original database schema.
-	for _, pt := range proposedProjection.Inference.Types {
-		if !SliceContains(pt, existingProjection.Inference.Types) {
+	for _, pt := range effectiveJsonTypes(proposedProjection) {
```

Comment:
nit: dropped comment seems useful?

Reply:
Yep I think I killed it on accident...brought it back with some modification.